apis/refs/v1beta1/dataexchangeref.go

```diff
@@ -115,3 +115,36 @@ func ResolveDataExchangeRef(ctx context.Context, reader client.Reader, obj clien
 		DataExchangeID: resourceID,
 	}, nil
 }
+
+func ResolveDataExchangeForObject(ctx context.Context, reader client.Reader, obj *unstructured.Unstructured) (*DataExchange, error) {
```

Comment:
So we are removing functions from apis/refs and restructure the resource reference. Adding more functions here would make it harder to finish the reference change. Therefore,  could you move this to the apis/bigqueryanalyticshub using the `ExternalNormalizer` interface?